data/provider/src/main/java/tech/pegasys/teku/api/ValidatorDataProvider.java

```diff
@@ -141,6 +165,28 @@ private SafeFuture<UInt256> retrieveExecutionPayloadValue(final UInt64 slot) {
         .orElseThrow(() -> new IllegalStateException("Execution Payload Value is not available"));
   }
 
+  private SafeFuture<UInt256> retrieveConsensusBlockRewards(final BlockContainer blockContainer) {
+    return combinedChainDataClient
+        .getStateAtSlotExact(blockContainer.getBlock().getSlot().decrement())
+        .thenApply(
+            maybeParentState ->
+                maybeParentState.map(
+                    state -> rewardCalculator.getBlockRewardData(blockContainer, state)))
+        .thenApply(
+            blockRewardData ->
+                blockRewardData.map(
+                    rewardData -> EthConstants.GWEI_TO_WEI.multiply(rewardData.getTotal())))
```

Comment:
Could we add some comments to `BlockRewardData` that everything is in GWei? (except proposer index)